mlflow/xgboost.py

```diff
@@ -257,8 +257,27 @@ def log_model(
 def _load_model(path):
     import xgboost as xgb
 
-    model = xgb.Booster()
-    model.load_model(os.path.abspath(path))
+    if os.path.isfile(path):
+        # xgboost Booster models saved in MLflow (<x.x.x) specify
+        # the ``data`` field within its flavor configuration.
+        # For these models, the ``path`` parameter of ``_load_pyfunc()``
+        # refers directly to a Booster model object.
+        # In this case, we create a Booster() instance and load model weights.
+        model = xgb.Booster()
+        model.load_model(os.path.abspath(path))
+    else:
+        # In contrast, xgboost models saved in new MLflow (>=x.x.x) do not
+        # specify the ``data`` field within its flavor configuration.
+        # We use ``model_class`` to specify its sklearn model class.
+        # For these models, the ``path`` parameter of ``load_pyfunc()``
+        # refers to the top-level MLflow Model directory.
+        # In this case, we first get the xgboost sklearn model from
+        # its flavor configuration and then create an instance based on its class.
+        flavor_conf = _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME)
+        model_class = flavor_conf.get("model_class", "Booster")
+        model_path = os.path.join(path, "model.xgb")
+        model = getattr(xgb, model_class)()
```

Comment:
Hi @dbczumar, we can definitely save the fully-qualified class name. Just a short question: We can save `xgboost.XGBRegressor` or `xgboost.sklearn.XGBRegressor`. Which one is preferred? Since both are valid: 
```
import xgboost
regressor = xgboost.XGBRegressor(args)
# also valid
regressor = xgboost.sklearn.XGBRegressor(args)
```

Reply:
On a second though, maybe `xgboost.sklearn.XGBRegressor` is better. It specifies the class is a sklearn model.